js/form/form.js

```diff
@@ -73,7 +76,49 @@ const setOfferValidation = (offerForm, offerPristineObject, formElementList) =>
   onCheckTimeChangeListener(formElementList.checkOut, formElementList.checkIn);
 };
 
-const prepareOfferForm = (offerForm, successPopup, errorPopup) => {
+const checkUploadFile = (file) => {
+  const fileName = file.name.toLowerCase();
+  return fileTypes.some((it) => fileName.endsWith(it)) ;
+};
+
+const avatarUploadListener = (inputTarget, previewTarget) => {
+  inputTarget.addEventListener('change', () => {
+    const inputFile = inputTarget.files[0];
+    if (checkUploadFile(inputFile)) {
+      previewTarget.src = URL.createObjectURL(inputFile);
+    }
+  });
+};
+
+const prepareImgElement = () => {
+  const imageElement = document.createElement('img');
+  imageElement.setAttribute('width', offerPreviewSetting.WIDTH);
+  imageElement.setAttribute('height', offerPreviewSetting.HEIGHT);
+  imageElement.setAttribute('alt', offerPreviewSetting.alt);
+  return imageElement;
+};
```

Comment:
Интересно, за какие заслуги высота и ширина удостоились капитализации, и чем провинился альтернативный текст?
Вообще нормальное решение, казалось бы. А если вот так? Будет ли универсальнее? Можно ли будет при других условиях использовать такую фабрику?

```suggestion
const prepareImgElement = ({
    width = offerPreviewSetting.WIDTH,
    height = offerPreviewSetting.HEIGHT,
    alt = offerPreviewSetting.alt,
}) => {
  const imageElement = document.createElement('img');
  imageElement.setAttribute('width', width);
  imageElement.setAttribute('height', height);
  imageElement.setAttribute('alt', alt);
  return imageElement;
};
```